superset/views/core.py

```diff
@@ -1038,6 +1038,18 @@ def get_viz(
             )
             return viz_obj
 
+    @has_access
+    @expose('/sql')
+    def sql_from_form_data(self):
```

Comment:
I feel like this endpoint already exists, let me check.

Reply:
Basically call `getExploreUrlAndPayload(formData, endpointType = true)`

Reply:
FYI it gets called when opening the "View Query" while the query is still running or has never run yet.